test/common.py

```diff
@@ -204,7 +205,11 @@ def tmp_digits_dataset(prefix: str,
                         start_pos = 0
                         end_pos = min(target_len, 3)
                         constraint = ' '.join(target_words[start_pos:end_pos])
-                        new_source['constraints'] = [constraint]
+                        # Alternate between positive and negative constraints
```

Comment:
should we add the case of both here as well?

Reply:
I think we should keep them separate, just because I don't want to always test both.

Reply:
ok, sure. We could create a third branch add either constraints, avoid, or both.